crates/uv-interpreter/src/python_query.rs

```diff
@@ -144,130 +73,411 @@ pub fn find_requested_python(
 /// `python.exe` respectively.
 #[instrument]
 pub fn find_default_python(platform: &Platform, cache: &Cache) -> Result<Interpreter, Error> {
-    let current_dir = env::current_dir()?;
-    let python = if cfg!(unix) {
-        which::which_in("python3", env::var_os("UV_TEST_PYTHON_PATH"), current_dir)
-            .or_else(|_| which::which("python3"))
-            .or_else(|_| which::which("python"))
-            .map_err(|_| Error::NoPythonInstalledUnix)?
-    } else if cfg!(windows) {
-        // TODO(konstin): Is that the right order, or should we look for `py --list-paths` first? With the current way
-        // it works even if the python launcher is not installed.
-        if let Ok(python) = which::which_in(
-            "python.exe",
-            env::var_os("UV_TEST_PYTHON_PATH"),
-            current_dir,
-        )
-        .or_else(|_| which::which("python.exe"))
-        {
-            python
-        } else {
-            installed_pythons_windows()?
+    find_python(PythonVersionSelector::Default, platform, cache)?.ok_or(Error::NoPythonInstalled)
+}
+
+fn find_python(
+    selector: PythonVersionSelector,
+    platform: &Platform,
+    cache: &Cache,
+) -> Result<Option<Interpreter>, Error> {
+    #[allow(non_snake_case)]
+    let UV_TEST_PYTHON_PATH = env::var_os("UV_TEST_PYTHON_PATH");
```

Comment:
I intentionally removed the TEST early return. We want that as much code as possible in this function is run during tests.